zebra-chain/src/transaction/serialize.rs

```diff
@@ -253,8 +254,14 @@ impl<P: ZkSnarkProof> ZcashSerialize for JoinSplit<P> {
 impl<P: ZkSnarkProof> ZcashDeserialize for JoinSplit<P> {
     fn zcash_deserialize<R: io::Read>(mut reader: R) -> Result<Self, SerializationError> {
         Ok(JoinSplit::<P> {
-            vpub_old: reader.read_u64::<LittleEndian>()?,
-            vpub_new: reader.read_u64::<LittleEndian>()?,
+            vpub_old: reader
+                .read_u64::<LittleEndian>()?
+                .try_into()
+                .map_err(|msg| io::Error::new(io::ErrorKind::Other, msg))?,
+            vpub_new: reader
+                .read_u64::<LittleEndian>()?
+                .try_into()
+                .map_err(|msg| io::Error::new(io::ErrorKind::Other, msg))?,
```

Comment:
I'm a little confused since this `try_into` is the conversion into an `Amount`, right (?), but the error is being mapped to an `io::Error` even though there's a `SerializationError::Amount`.  Should it be using that error type instead, or is there a reason that it should be an `io::Error`?

Reply:
iirc this function returns an `io::Error` we could map to a `SerializationError::Amount` first but then we'd still have to map to an `io::Error` unless we want to update the fn signature, let me take a look.

Reply:
nope i got it wrong, this should be a SerializationError, dont know how I got this one mixed up